app/controllers/vaccines_controller.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+# VaccinesController: patient immunizations
+class VaccinesController < ApplicationController
+  include VaccineQueryHelper
+
+  before_action :authenticate_user!
+
+  def index
+    redirect_to(root_url) && return unless current_user&.can_view_patient_vaccines?
+
+    # Validate params and handle errors if invalid
+    begin
+      data = validate_table_query(params)
+    rescue StandardError => e
+      render(json: { error: e.message }, status: :bad_request) && return
+    end
+
+    # Verify user has access to patient, patient exists, and the patient has vaccines
+    patient = current_user.get_patient(data[:patient_id])
+    vaccines = patient&.vaccines
+    redirect_to(root_url) && return if patient.nil? || vaccines.blank?
```

Comment:
Is `vaccines.blank?` going to ever redirect just because the patient doesn't have any vaccinations? That doesn't seem to be the case when I load a patient page without any vaccinations but when I do `Patient.first.vaccines.blank?`, it returns `true`. This probably isn't a problem, but I'm just trying to understand what's going on here

Reply:
This line is hit when the vaccines are blank. It's not actually redirecting the user ultimately because the request made to get these vaccinations is made asynchronously from a React component which doesn't handle the 302 redirects, so it is just ignored.

Reply:
got it, thanks for the explanation!